main.go

```diff
@@ -1,68 +1,160 @@
 package main
 
 import (
-	"os"
-	"fmt"
-	"net"
-	"strings"
+	"bufio"
+	"flag"
+	"io"
+	"log"
 	"math/rand"
-	"io/ioutil"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
 )
 
-func randomQuote() ([]byte, error) {
-	bs, err := ioutil.ReadFile(os.Args[2])
+type quoteServer struct {
```

Comment:
Whenever you have a handler that has "state" it is usually a good idea to make that handler a type

Reply:
Why?

Reply:
So you have somewhere to store it. In this case, previously holding state was avoided by re-reading the quote file upon every request. Using the `quoteServer` type allows us to store the parsed quotes in memory and do less work to respond to a request.

Reply:
I see. So the `quoteServer` type is only instantiated once and the actual handling is done by calling methods on it, correct?

Reply:
Yes, it is possible to reload the quotes file given when the program is started by sending the SIGHUP signal however.